### PR TITLE
Added new Java 9 methods, or and ifPresentOrElse.

### PR DESCRIPTION
**Pull request by fieldju**
Implement features requested in issue: https://github.com/JasonStorey/Optional.js/issues/14

Implemented [Optional::ifPresentOrElse](https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#ifPresentOrElse-java.util.function.Consumer-java.lang.Runnable-) and [Optional::or](https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#or-java.util.function.Supplier-) as described in the [Java 9 Optional API Doc](https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html)

### CODE REVIEW

**Comment by fieldju** [comment]
So the only thing I left out was https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#stream--

I'm not sure how to go about that, or if that is a thing you would do in javascript.

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Optional.ofNullable(promptForUserName)
 
 # Features
 
-- Full Java 8 Optional API is supported
+- Full Java 9 Optional API is supported
```
Comment (fieldju) [inline README.md]: If stream is left out them maybe this should be something like?

```
Full Java 8 API and partial Java 9 API is supported
```

File: test/optional-test.js [hunk 2]
```diff
@@ -311,5 +311,64 @@ describe('Optional.js', function () {
...
         });
+
+        it('.ifPresentOrElse() if value is present calls the consumer and not the else callable', function () {
+            var consumerCalled = false,
+                elseCallableCalled = false,
+                passedValue;
+
+            nonNullOptional.ifPresentOrElse(
+                function consumer(value) {
+                    consumerCalled = true;
```
Comment (JasonStorey) [inline test/optional-test.js]: I should probably add Sinon & Mocha - a decent stubbing lib would make these tests much less hassle to write

**Comment by JasonStorey** [comment]
@fieldju I've added an issue to continue discussion on Optional::stream() method - https://github.com/JasonStorey/Optional.js/issues/16

**Pull request merged**